### PR TITLE
R11PIT-775 - Update hosting bundle from .NET Core 3.1 to .NET 6.0

### PR DESCRIPTION
**Pull request by OS-tiagobarroso**
This PR is related with [R11PIT-775](https://outsystemsrd.atlassian.net/browse/R11PIT-775). 

New changes were made in order to upgrade the hosting bundle to .NET 6.0 on newer Platform Server versions (>= 11.17.1). Some new tests were also added to confirm that the behavior is the expected one.

### CODE REVIEW

**Review by nunojrsilva — DISMISSED** [review]

File: src/Outsystems.SetupTools/Functions/Install-OSServerPreReqs.ps1
Comment (OS-tiagobarroso) [reply src/Outsystems.SetupTools/Functions/Install-OSServerPreReqs.ps1]: Done, but need you approval again 😛

**Review by pedro132 — DISMISSED** [review]

**Review by nunojrsilva — DISMISSED** [review]

**Comment by luisalourenco** [comment]
Are you aware of this documentation https://outsystemsrd.atlassian.net/wiki/spaces/RKB/pages/1037861120/How+to+change+the+version+of+Outsystems.SetupTools ? There is a missing step I think regarding changelog

**Comment by OS-tiagobarroso** [comment]
I was not aware, but regarding that topic, the update version process appears to have been done in separate commits previously. Do we want do the as that guide suggests or how it has been done previously?

**Comment by luisalourenco** [comment]
I need to align with Pedro Antunes regarding how releases are made using GitFlow (Platform uses GitHub Flow) and with TAF team to understand and align their expectations since they are also making changes we do not want to bundle together.
This PR iis on hold until these alignments are concluded

**Review by luisalourenco — APPROVED** [review]

**Pull request merged**
